hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java

```diff
@@ -76,7 +76,8 @@ public void prepareRun(BatchSinkContext context) throws Exception {
   public void transform(StructuredRecord input, Emitter<KeyValue<Text, NullWritable>> emitter) throws Exception {
     List<String> dataArray = new ArrayList<>();
     for (Schema.Field field : input.getSchema().getFields()) {
-      dataArray.add(input.get(field.getName()).toString());
+      String data = (input.get(field.getName()) != null) ? input.get(field.getName()).toString() : "";
```

Comment:
no need to get the field from the input twice.
Would be better to do the lookup just once, and reuse it from a local var